indico/modules/events/sessions/controllers/display.py

```diff
@@ -34,3 +37,51 @@ def _process(self):
         sessions = get_sessions_for_user(self.event_new, session.user)
         return WPDisplayMySessionsConference.render_template('display/session_list.html', self._conf,
                                                              event=self.event_new, sessions=sessions)
+
+
+class RHDisplaySessionBase(RHConferenceBaseDisplay):
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.session
+        }
+    }
+
+    def _checkParams(self, params):
+        RHConferenceBaseDisplay._checkParams(self, params)
+        self.session = self.event_new.sessions.filter_by(id=request.view_args['session_id'], is_deleted=False).one()
```

Comment:
Since you use normalization you can just use `Session.get(request.view_args['session_id'] is_deleted=False)